api/v1alpha1/nstemplatetier_types.go

```diff
@@ -26,6 +26,24 @@ type NSTemplateTierSpec struct {
 	// SpaceRequestConfig stores all the configuration related to the Space Request feature
 	// +optional
 	SpaceRequestConfig *SpaceRequestConfig `json:"spaceRequestConfig,omitempty"`
+
+	// Parameters is an optional array of Parameters used during the NSTemplateTier and TierTemplate creation.
+	// When creating the NsTemplateTier and referenced TierTemplates, the parameters will be read from the NSTemplateTier and evaluated in all the TierTemplates referenced in the spec.
+	// +optional
+	// +listType=atomic
+	Parameters []Parameter `json:"parameters,omitempty" protobuf:"bytes,4,opt,name=parameters"`
```

Comment:
Since the name of each parameter is unique, it's a perfect fit for using `map` instead of `atomic`:
```suggestion
	// +optional
	// +patchMergeKey=name
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=name
	Parameters []Parameter `json:"parameters,omitempty" protobuf:"bytes,4,opt,name=parameters" patchStrategy:"merge" patchMergeKey:"name"`
```

Reply:
good point! Thanks!